### PR TITLE
fix: restrict python version and update cattrs version restriction

### PR DESCRIPTION
**Pull request by alekseiloginov**
## Description

We need to unpin `cattrs` version to avoid dependency conflicts with other tools using a later version of it. 

`cattrs` version was [pinned](https://github.com/looker-open-source/sdk-codegen/commit/bf8f3f694efd3a2b7af7d5c849694e0c7bd399c5) to `1.1.2` due to https://github.com/Tinche/cattrs/issues/119. Since the issue should be [fixed](https://github.com/python-attrs/cattrs/issues/119#issuecomment-789757914) in `cattrs` `1.3`, we update the restriction to `>=1.3`.

Also `python` version need to be restricted to `<=3.9.9` due to https://github.com/looker-open-source/sdk-codegen/issues/944. Since `Pipfile` doesn't support qualifiers for `python_full_version`, it's set to `3.9.9`.

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -38,7 +38,6 @@
     "python-dateutil;python_version<'3.7'",
     # Python 3.7+
     "attrs >= 20.1.0;python_version>='3.7'",
-    "cattrs == 1.1.2;python_version>='3.7'",
```
Comment (joeldodge79) [inline python/setup.py:41]: oops, actually we need to still specify a cattrs dep here ([ci failed](https://github.com/looker-open-source/sdk-codegen/runs/4902006917?check_suite_focus=true)).

```
cattrs >= 1.3;python_version>='3.7'
```

should have the fix?

**Review by joeldodge79 — CHANGES_REQUESTED** [review]

**Comment by joeldodge79** [comment]
also, lets change the title to something like the following (to fit conventional commit msg standard)

`fix: restrict python version and remove cattrs version`

File: python/setup.py (hunk 1 above)
Comment (alekseiloginov) [reply python/setup.py:41]: Done, also updated `Pipfile` with `cattrs = ">=1.3"`.

**Review by jkaster — APPROVED** [review]
LGTM

**Review by joeldodge79 — APPROVED** [review]

**Comment by potiuk** [comment]
Just one comment here - pinning python version in a library is a very bad idea. What should people do if they already have python 3.9.10 installed ? This - for example - prevents https://github.com/apache/airflow/pull/20882 from merging.

**Comment by potiuk** [comment]
The consequence of this change is that any other package that would be released with looker SDK released with this limitation will not be installable not only on Python 3.9.10 but also any other version it follows. This is extremely bad practice to upper-bound a Python version IMHO. I have never, ever seen a library doing it. It might be fine for an application if it has specific requirements and they are broken (because that application might simply be ok to refuse working on specific python version) but restricting an SDK/library to not being installable on a python version means that you are preventing any other packages that use looker SDK in this version from even installing on newer Python versions.

**Comment by joeldodge79** [comment]
@potiuk you're correct, it's a less than ideal situation. We're working to resolve the issue but the choice is between pinning it ourselves or letting users find out that it's broken on 3.9.10+

The PR title should have reflected that this is a temporary measure to choose between the lesser of two evils while we find the resources to fix the actual issue

**Pull request merged**